PhotoFeed/app/src/main/java/com/example/yhuang/photofeed/MainActivity.java

```diff
@@ -0,0 +1,212 @@
+        search = (Button) findViewById(R.id.button);
+        pre = (Button) findViewById(R.id.pre_btn);
+        next = (Button) findViewById(R.id.next_btn);
+        feed = (Button) findViewById(R.id.feed_btn);
+        delete = (Button) findViewById(R.id.delete);
+        imageview = (ImageView) findViewById(R.id.imageView);
+        mDbHelper = new PhotoHelper(this);
+
+        //open the search page
+        openSearch();
+
+        //set onClickListeners to the buttons
+        search.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                //Get the search text in edit text
+                String searchText = editText.getText().toString();
+                makeRequestWithCallback(searchText, items);
+                editText.setText("");
+            }
+        });
+
+        pre.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (!items.isEmpty() && cur_page > 0) {
+                    cur_page -= 1;
+                    new ImageLoadTask(items.get(cur_page), imageview).execute();
+                } else {
+                    Toast.makeText(getApplicationContext(), "Already the first page.", Toast.LENGTH_SHORT).show();
+                }
+            }
+        });
+
+        next.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (!items.isEmpty() && cur_page < items.size() - 1) {
+                    cur_page += 1;
+                    new ImageLoadTask(items.get(cur_page), imageview).execute();
+                } else {
+                    Toast.makeText(getApplicationContext(), "Already the last page.", Toast.LENGTH_SHORT).show();
+                }
+            }
+        });
+
+        feed.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (!items.isEmpty()) {
+                    PhotoHelper.WriteIntoDatabase(mDbHelper, items.get(cur_page));
+                    Toast.makeText(getApplicationContext(), "Added to feed", Toast.LENGTH_SHORT).show();
+                } else {
+                    Toast.makeText(getApplicationContext(), "Nothing to feed", Toast.LENGTH_SHORT).show();
+                }
+            }
+        });
+
+        delete.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (!items.isEmpty()) {
+                    PhotoHelper.DeleteItemDatabase(mDbHelper, items.get(cur_page));
+                    Toast.makeText(getApplicationContext(), "Deleted from feed", Toast.LENGTH_SHORT).show();
+                    openMyFeed();
+                } else {
+                    Toast.makeText(getApplicationContext(), "Nothing to delete", Toast.LENGTH_SHORT).show();
+                }
+            }
+        });
+    }
```

Comment:
Moving the onclicklisteners to a seperate file or class could be useful for clarity. Also nice use of toasts.